### ISSUE TITLE
`no-extra-non-null-assertion` doesn't like non-null assertions after optional chaining operator

### ISSUE THREAD

**Issue by greentore** [denoland/deno_lint#1432]
### Lint Name

no-extra-non-null-assertion

### Code Snippet

```ts
document.children[0]?.textContent!.trim();
```

### Expected Result

No complains about non-null assertion after `textContent`.

### Actual Result

Yellow squiggle with `!`, red squiggle without.

### Version

deno 2.2.11 (stable, release, x86_64-pc-windows-msvc)
v8 13.5.212.10-rusty
typescript 5.7.3
